src/main/java/de/tobiasroeser/lambdatest/proxy/TestProxy.java

```diff
@@ -223,7 +223,8 @@ private static String mkArgListString(Method method) {
 			final boolean isErasedOrObject = Object.class.equals(clazz);
 
 			final String clazzSimpleName = clazz.getSimpleName();
-			final String argName = decapitalize(clazzSimpleName);
+			final String argName1 = decapitalize(clazzSimpleName).replaceAll("[^a-zA-Z0-1_]+", "");
+			final String argName = argName1.length() == 0 ? "x" : argName1;
```

Comment:
The length check shouldn't be necessary.

Reply:
The class names can be more complex https://stackoverflow.com/questions/65475/valid-characters-in-a-java-class-name e.g. an unicode character, i tried to be on the safe side here.
After the whitelist filtering an empty string can happen.

Reply:
we could use a better argName as "x" for this case, what do you think?

Reply:
Ok, then I'm in favor of a blacklist for the special case `[]`, as rules for class names and parameter names should be identical.

Reply:
done in next commit 4041c92.